### DISCUSSION TITLE
What's the best way to wait for network after a power outage?

### DISCUSSION THREAD

**Discussion by lazyoldjack** [toitlang/toit#771] [Q&A]
I realise I could handle errors etc. in my program, but is there an easy way to not start my program before Toit has internet? Toit is up in a few seconds and WiFi internet connection takes a few minutes here, while the modem cranks up after a power outage. I looked at the yaml specifications but couldn't see an appropriate option.

**Comment by floitsch** [comment]
I don't think there is any option for that.
@kasperl : a use case to keep in mind.

**Comment by lazyoldjack** [comment]
Interestingly, sometimes it keeps resetting if network not available, 
which I guess comes to the same thing as wait for network, but it 
doesn't always do this. See yellow highlight in pic below.

Also, oddly, it reset and connected to the WiFi when it was able to, but 
then reset and connected two more times. See blue highlight in picture 
below. Then it settled down and my program ran.

Sorry for the screen grab; how do I get the logs in a usable format? I 
tried scraping but your (over?) complex page defeated me. Whatever 
happened to <table>...</table> :oD I guess you can't keep adding rows as 
people scroll, so I do get it, but a button to spit out a log might be 
useful; did I miss it?

Anyway, thought this feedback might be relevant useful, cheers, J.

![log1](https://user-images.githubusercontent.com/20986290/172934036-cf1fbda8-0931-4506-b8cb-e0c12c8d8175.png)


On 07/06/2022 13:28, Florian Loitsch wrote:
>
> I don't think there is any option for that.
> @kasperl <https://github.com/kasperl> : a use case to keep in mind.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/toitlang/toit/discussions/771#discussioncomment-2897663>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AFADTMWZZICMNT7MF2SZ2J3VN455DANCNFSM5X6WBSSQ>.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>

-- 
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

**Reply by kasperl** [reply]
The logic for running the app is pretty complicated. If we cannot connect a number of times, we worry that it might be the app that is interfering with that, so we stop running it until we connect to the cloud again. Between attempts we go to sleep (deep sleep) for about 5s if max-offline is 0, so you get a lot of soft restarts of the device until we connect again. If you're running with max-offline > 0, we will also lower the setting to get more frequent connect attempts (we attempt 4 times as frequently).

You can subscribe to logs from your devices like this:

```
$ toit data subscription create logs my-logs
$ toit data read logs my-logs --auto-ack --output table
```

You can experiment with the different output formats (json might be useful).

**Reply by lazyoldjack** [reply]
OK, thanks for the explanation! Will try subscribing to the logs...
